dashboard/app/helpers/aitutor_system_prompt_helper.rb

```diff
@@ -0,0 +1,81 @@
+      this topic, please say: \"I'm sorry, but I can't assist with that.\"
+      Inappropriate Language: Immediately disengage from conversations that include swear words or disrespectful language, saying,
+      \"I'm sorry, but I can't assist with that.\"
+      Cultural Sensitivity: Approach all discussions with cultural sensitivity and an awareness of diverse perspectives, ensuring inclusivity.
+      Encouragement and Positivity: Foster a positive learning environment, encouraging questions and curiosity within the educational scope.
+      Safeguarding Student Well-being: Privacy and Confidentiality: Ensure discussions respect student privacy and confidentiality,
+      avoiding personal or sensitive topics not related to computer science. Responses to content that fall outside the scope of what
+      is allowed should be responded to with: \"I'm sorry, but I can't assist with that.\"
+      Overall Objective: This comprehensive framework aims to cultivate a focused, respectful, and enriching educational dialogue within a
+      computer science classroom, prioritizing student learning, safety, and inclusivity.
+      If you determine the question is relevant to computer science, remember:
+      1. The student prefers being guided to discover the solution themselves over being directly provided with answers.
+      2. The student prefers to receive guided questions that prompt them to reevaluate assumptions or scrutinize overlooked details in their code.
+      3. The student is interested in being reminded of important principles and concepts in programming, which can often be the root cause of issues in their code.
+      Direct Support: Engage exclusively in discussions that directly support the Code.org computer science curriculum,
+      focusing solely on the context of a specific level or lesson. Your job is to provide the student with one next step to take, in priority order:
+      If their code does not compile, give them a hint about why
+      If their tests are failing, give them a hint about why
+      If their solution is missing something, give them a hint about what it's missing.
+      If there is a bug in their code, tell them where the issue is."
+    base_system_prompt
+  end
+
+  def self.get_language_specific_system_prompt(unit)
+    language = unit.csa? ? 'Java' : 'Python'
+    "\n Specific Exclusions: Refrain from discussing topics not explicitly related to computer
+    science or #{language} programming."
+  end
+
+  def self.get_level_instructions(level)
+    level_instructions = level.properties["long_instructions"]
+    "\n Here are the student instructions for this level: #{level_instructions}"
+  end
+
+  def self.get_validated_level_test_file_contents(level)
+    test_file_contents = ""
+    if level.validation
+      if level.validation.values.empty?
+        return render(status: :bad_request, json: {message: "There are no test files associated with level id=#{level_id}."})
```

Comment:
For my understanding, why don't we return an error if a level has no validation (but do if its values are empty)?

Reply:
My understanding that it's expected that there would be non-validated JavaLab levels, but if the level _is_ validated and we can't find the test files then something went wrong.

Reply:
Got it, thanks!